src/GitHub.UI/Controls/ComboBoxes/LinkDropDown.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+
+namespace GitHub.UI
+{
+    /// <summary>
+    /// A ComboBox that displays as a link with a dropdown.
+    /// </summary>
+    public class LinkDropDown : ComboBox
+    {
+        /// <summary>
+        /// Defines the <see cref="LinkText"/> property.
+        /// </summary>
+        static readonly DependencyPropertyKey LinkTextPropertyKey =
+            DependencyProperty.RegisterReadOnly(
+                "LinkText",
+                typeof(string),
+                typeof(LinkDropDown),
+                new FrameworkPropertyMetadata(string.Empty));
+
+        /// <summary>
+        /// Defines the <see cref="Header"/> property.
+        /// </summary>
+        public static readonly DependencyProperty HeaderProperty =
+            HeaderedItemsControl.HeaderProperty.AddOwner(
+                typeof(LinkDropDown),
+                new FrameworkPropertyMetadata(typeof(LinkDropDown), HeaderChanged));
+
+        /// <summary>
+        /// Defines the readonly <see cref="LinkText"/> property.
+        /// </summary>
+        public static readonly DependencyProperty LinkTextProperty =
+            LinkTextPropertyKey.DependencyProperty;
+
+        /// <summary>
+        /// Initializes static members of the <see cref="LinkDropDown"/> class.
+        /// </summary>
+        static LinkDropDown()
+        {
+            DefaultStyleKeyProperty.OverrideMetadata(
+                typeof(LinkDropDown),
+                new FrameworkPropertyMetadata(typeof(LinkDropDown)));
+        }
+
+        /// <summary>
+        /// Gets or sets a header to use as the link text when no item is selected.
+        /// </summary>
+        public object Header
+        {
+            get { return GetValue(HeaderProperty); }
+            set { SetValue(HeaderProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the text to display in the link.
+        /// </summary>
+        public string LinkText
+        {
+            get { return (string)GetValue(LinkTextProperty); }
+            private set { SetValue(LinkTextPropertyKey, value); }
+        }
```

Comment:
If any of these properties are allowed to be set to null, you need to add the `[AllowNull]` attribute to them, since NullGuard will add exception throwing code otherwise.
If they'll have a null value at some point (either initially or later in the lifetime of the instance), then the getter needs a `[return: AllowNull]` attribute.